datasets/austin_bikeshare/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,167 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import json
+import logging
+import math
+import os
+import pathlib
+import typing
+
+import pandas as pd
+import requests
+from google.cloud import storage
+
+
+def main(
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+    headers: typing.List[str],
+    rename_mappings: dict,
+    pipeline_name: str,
+) -> None:
+
+    logging.info(
+        f"Austin bikeshare {pipeline_name} process started at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+    logging.info("creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+
+    logging.info(f"Downloading file from {source_url}...")
+    download_file(source_url, source_file)
+
+    # open the input file
+    logging.info(f"Opening file {source_file}...")
+    df = pd.read_csv(str(source_file))
+
+    # steps in the pipeline
+    logging.info(f"Transforming {source_file}... ")
+
+    logging.info(f"Transform: Rename columns.. {source_file}")
+    rename_headers(df, rename_mappings)
+
+    logging.info(f"Transform: Converting date values for {source_file}... ")
+    df["modified_date"] = df["modified_date"].apply(convert_dt_format)
+
+    logging.info(f"Transform: filtering rows for {source_file}... ")
+    filter_null_rows(df)
+
+    logging.info(f"Transform: converting to integer {source_file}... ")
+    df["city_asset_number"] = df["city_asset_number"].apply(convert_to_int)
+    df["number_of_docks"] = df["number_of_docks"].apply(convert_to_int)
+    df["footprint_length"] = df["footprint_length"].apply(convert_to_int)
+    df["council_district"] = df["council_district"].apply(convert_to_int)
+
+    logging.info(f"Transform: cremoving NaN values {source_file}... ")
+    df["footprint_width"] = df["footprint_width"].apply(resolve_nan)
+
+    logging.info("Transform: Reordering headers..")
+    df = df[headers]
+
+    logging.info(f"Saving to output file.. {target_file}")
+    try:
+        save_to_new_file(df, file_path=str(target_file))
+    except Exception as e:
+        logging.error(f"Error saving output file: {e}.")
+
+    logging.info(
+        f"Uploading output file to.. gs://{target_gcs_bucket}/{target_gcs_path}"
+    )
+    upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+    logging.info(
+        f"Austin bikeshare {pipeline_name} process completed at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+
+def resolve_nan(input: str) -> str:
+    str_val = ""
+    if input == "" or (math.isnan(input)):
```

Comment:
Because you're applying `isnan`, `input` is expected to be either a string or a float (?). 

You can use  `resolve_nan(input: typing.Union[str, float])`

Reply:
changed